src/client/datascience/export/exportManagerDependencyChecker.ts

```diff
@@ -11,19 +12,23 @@ export class ExportManagerDependencyChecker implements IExportManager {
         @inject(ExportManager) private readonly manager: IExportManager,
         @inject(IJupyterExecution) private jupyterExecution: IJupyterExecution,
         @inject(IJupyterInterpreterDependencyManager)
-        private readonly dependencyManager: IJupyterInterpreterDependencyManager
+        private readonly dependencyManager: IJupyterInterpreterDependencyManager,
+        @inject(ProgressReporter) private readonly progressReporter: ProgressReporter
     ) {}
 
     public async export(format: ExportFormat, model: INotebookModel): Promise<Uri | undefined> {
         // Before we try the import, see if we don't support it, if we don't give a chance to install dependencies
-        if (!(await this.jupyterExecution.isImportSupported())) {
-            await this.dependencyManager.installMissingDependencies();
-        }
-
-        if (await this.jupyterExecution.isImportSupported()) {
-            return this.manager.export(format, model);
-        } else {
-            throw new Error(localize.DataScience.jupyterNbConvertNotSupported());
+        const reporter = this.progressReporter.createProgressIndicator(`Exporting to ${format}`);
+        try {
+            if (!(await this.jupyterExecution.isImportSupported())) {
+                await this.dependencyManager.installMissingDependencies();
+            }
+            if (!(await this.jupyterExecution.isImportSupported())) {
+                throw new Error(localize.DataScience.jupyterNbConvertNotSupported());
+            }
```

Comment:
```suggestion
            if (!(await this.jupyterExecution.isImportSupported())) {
                await this.dependencyManager.installMissingDependencies();
	            if (!(await this.jupyterExecution.isImportSupported())) {
    	            throw new Error(localize.DataScience.jupyterNbConvertNotSupported());
        	    }
        	}
```

This way we check it a second time only if the first time failed.

Reply:
I.e. avoid duplicate checks if it is installed (considering this check is spawning processes we should try to avoid if possible)